src/charm.py

```diff
@@ -50,10 +51,11 @@ def _on_relation_created(self, event: ops.RelationCreatedEvent) -> None:
         """
         if not self.model.unit.is_leader():
             return
-        if secret_id := self._charm_state.password_id:
-            secret = self.model.get_secret(id=secret_id)
-            secret.grant(event.relation)
-        self._update_smtp_relation(event.relation)
+        if self._has_secrets():
+            secret = self._store_password_as_secret()
+            if secret:
```

Comment:
What happens if is None?

Reply:
It will happen when no password is set. The condition is there so that the `grant` won't break